shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/RouteCardData.kt

```diff
@@ -195,14 +195,17 @@ data class RouteCardData(
             now: Instant,
             representativeRoute: Route,
             globalData: GlobalResponse?,
+            context: Context,
         ): Set<PotentialService> {
             val potentialService: MutableMap<Pair<String, String>, MutableSet<String>> =
                 mutableMapOf()
             val cutoffTime = now + 120.minutes
             val tripsUpcoming = upcomingTrips.filter { it.isUpcomingWithin(now, cutoffTime) }
             val isBus = representativeRoute.type == RouteType.BUS
             val tripsToConsider =
-                if (isBus) tripsUpcoming.take(TYPICAL_LEAF_ROWS) else tripsUpcoming
+                if (isBus && context == Context.StopDetailsFiltered) tripsUpcoming
+                else tripsUpcoming.take(TYPICAL_LEAF_ROWS)
```

Comment:
Actually, looking at this a little closer, is this behavior change on non-bus routes something that we’re sure we want?

Reply:
Ooh great catch